src/main/kotlin/org/rust/lang/core/resolve/ResolveEngine.kt

```diff
@@ -134,9 +134,20 @@ object ResolveEngine {
     fun resolveMethodCallExpr(call: RsMethodCallExpr): RsNamedElement? {
         val receiverType = call.expr.type
         val name = call.identifier.text
+        var method: RsFunction? = null
 
-        return receiverType.getMethodsIn(call.project)
-            .find { it.name == name }
+        receiverType.getMethodsIn(call.project)
```

Comment:
This `forEach` with early return and `var method` are not elegant, but I think it's Ok for the time being. I'd like to rethink API around trait-based resolve some day. https://gitter.im/intellij-rust/intellij-rust?at=586d8908058ca96737cc43ea has  interesting ideas.

But perhaps for the problem at hand we can do something like `methods.filter {it.name == name}.partition { it.isInherent }`?